packages/accounts/src/msca/multi-owner-account.ts

```diff
@@ -45,16 +50,24 @@ export const createMultiOwnerMSCABuilder = <
 
   const builder = new MSCABuilder()
     .withFactory(async (acct) => {
-      const ownerAddress = await params.owner.getAddress();
+      const ownerAddress = await params.signer.getAddress();
+      // owners need to be dedupe + ordered in ascending order and not == to zero address
+      const owners = Array.from(new Set([...params.owners, ownerAddress]))
+        .filter((x) => hexToBigInt(x) !== 0n)
+        .sort((a, b) => {
+          const bigintA = hexToBigInt(a);
+          const bigintB = hexToBigInt(b);
+
+          return bigintA < bigintB ? -1 : bigintA > bigintB ? 1 : 0;
+        });
```

Comment:
huh you sort address by changing to big int and sort in ascending order? why is that? is that how it's required on the contract?

Reply:
ah: https://ethereum.stackexchange.com/questions/125445/sort-address-uniswap-v2